Projects/Scripts/Misc/Email.cs

```diff
@@ -1,98 +1,154 @@
+    /// </summary>
+    /// <param name="filePath"></param>
+    public static void SendCrashEmail(string filePath)
     {
-      if (EmailServer != null)
+      //=========================================================[HEADER]
+      //=========================================================[HEADER]
+      var message = new MimeMessage();
+      message.To.Add(CrashAddresses);
+      message.Subject = "Automated RunUO Crash Report";
+      //=========================================================[BODY]
+      //=========================================================[BODY]
+      var builder = new BodyBuilder
       {
-        _Client = new SmtpClient(EmailServer, EmailPort);
-        if (EmailUsername != null) _Client.Credentials = new NetworkCredential(EmailUsername, EmailPassword);
-      }
+        TextBody = "Automated RunUO Crash Report. See attachment for details.",
+        HtmlBody = null
+      };
+      builder.Attachments.Add(filePath);
+      message.Body = builder.ToMessageBody();
+
+
     }
 
-    public static bool Send(MailMessage message)
+    /// <summary>
+    /// Sends emails async
+    /// </summary>
+    /// <param name="message"></param>
+    private static async void SendAsync(MimeMessage message)
     {
-      try
-      {
-        // .NET relies on the MTA to generate Message-ID header. Not all MTAs will add this header.
 
-        DateTime now = DateTime.UtcNow;
-        string messageID = $"<{now.ToString("yyyyMMdd")}.{now.ToString("HHmmssff")}@{EmailServer}>";
-        message.Headers.Add("Message-ID", messageID);
+      DateTime now = DateTime.UtcNow;
+      string messageID = $"<{now.ToString("yyyyMMdd")}.{now.ToString("HHmmssff")}@{EmailServer}>";
+      message.Headers.Add("Message-ID", messageID);
+      message.From.Add(FromAddress);
 
-        message.Headers.Add("X-Mailer", "RunUO");
-
-        lock (_Client)
+      for (var count = 1; count <= m_retryCount; count++)
+      {
+        try
         {
-          _Client.Send(message);
+          using (var client = new SmtpClient())
+          {
+
+            // For demo-purposes, accept all SSL certificates (in case the server supports STARTTLS)
+            client.ServerCertificateValidationCallback = (s, c, h, e) => true;
+            await client.ConnectAsync(EmailServer, EmailPort, false).ConfigureAwait(false);
+            await client.AuthenticateAsync(EmailServerUsername, EmailServerPassword);
+            await client.SendAsync(message).ConfigureAwait(false);
+            await client.DisconnectAsync(true).ConfigureAwait(false);
+            Console.WriteLine("Sent e-mail '{0}' to '{1}'.", message.Subject, message.To);
+            return;
+          }
+        }
+        catch (Exception exception)
+        {
+          Console.WriteLine(exception);
+          if (m_retryCount >= 0)
+          {
+            Console.WriteLine("Failure sending e-mail '{0}' to '{1}'.", message.Subject, message.To);
+            throw;

```

Comment:
This throw is generic. Not generally a good idea. This needs reworking due to the comment above too.